### PR TITLE
[SPARK-48357][SQL] Support for LOOP statement

### PR DESCRIPTION
**Pull request by dusantism-db**
### What changes were proposed in this pull request?
In this PR, support for LOOP statement in SQL scripting is introduced.

Changes summary:

Grammar/parser changes:
- `loopStatement` grammar rule
- `visitLoopStatement` rule visitor
- `LoopStatement` logical operator

`LoopStatementExec` execution node
Iterator implementation - repeatedly execute body (only way to stop the loop is with LEAVE, or if an exception occurs)
`SqlScriptingInterpreter` - added logic to transform LoopStatement logical operator to LoopStatementExec execution node

### Why are the changes needed?
This is a part of SQL Scripting introduced to Spark, LOOP statement is a basic control flow construct in the SQL language.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
New tests are introduced to scripting test suites: `SqlScriptingParserSuite`, `SqlScriptingExecutionNodeSuite` and `SqlScriptingInterpreterSuite`.

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/SqlScriptingLogicalOperators.scala [hunk 1]
```diff
@@ -138,3 +137,16 @@ case class CaseStatement(
...
+ * @param body Compound body is a collection of statements that are executed until the
+ *             LOOP statement is terminated by using the LEAVE statement.
+ * @param label An optional label for the loop which is unique amongst all labels for statements
+ *              within which the LOOP statement is contained.
+ *              If an end label is specified it must match the beginning label.
+ *              The label can be used to LEAVE or ITERATE the loop.
+ */
+case class LoopStatement(
+  body: CompoundBody,
+  label: Option[String]) extends CompoundPlanStatement
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/SqlScriptingLogicalOperators.scala:152]: please, fix indentations:
```suggestion
    body: CompoundBody,
    label: Option[String]) extends CompoundPlanStatement
```

**Review by MaxGekk — COMMENTED** [review]
LGTM except of a minor comment.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/SqlScriptingLogicalOperators.scala (hunk 1 above)
Comment (dusantism-db) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/SqlScriptingLogicalOperators.scala]: @MaxGekk fixed

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
The test failure of `AdaptiveQueryExecSuite` is not related to changes. I re-ran it locally, BTW.

+1, LGTM. Merging to master.
Thank you, @dusantism-db.

**Pull request closed without merging**